app/shared/app-data/src/commonMain/kotlin/data/persistent/database/dao/WebSearchEpisodeInfoDao.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2024 OpenAni and contributors.
+ *
+ * 此源代码的使用受 GNU AFFERO GENERAL PUBLIC LICENSE version 3 许可证的约束, 可以在以下链接找到该许可证.
+ * Use of this source code is governed by the GNU AGPLv3 license, which can be found at the following link.
+ *
+ * https://github.com/open-ani/ani/blob/main/LICENSE
+ */
+
+package me.him188.ani.app.data.persistent.database.dao
+
+import androidx.room.Dao
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.Index
+import androidx.room.PrimaryKey
+import androidx.room.Query
+import androidx.room.Transaction
+import androidx.room.Upsert
+import me.him188.ani.datasources.api.EpisodeSort
+
+@Entity(
+    tableName = "web_search_episode_info",
+    indices = [
+        Index(value = ["mediaSourceId", "subjectName"]),
+    ],
+    foreignKeys = [
+        ForeignKey(
+            entity = WebSearchSubjectInfoEntity::class,
+            parentColumns = ["mediaSourceId", "subjectName"],
+            childColumns = ["mediaSourceId", "subjectName"],
+            onDelete = ForeignKey.CASCADE,
+        ),
+    ],
+)
+data class WebSearchEpisodeInfoEntity(
+    @PrimaryKey(autoGenerate = true) val id: Int = 0,
```

Comment:
这个 entity 可以用 mediaSourceId, channel, namd 做主键吧

Reply:
channel 是 nullable 可以做主键吗

Reply:
不太清楚，试试？

Reply:
是不可以的

Reply:
playUrl可以做主键，但是把 id 换成 playUrl 意义似乎不是很大

Reply:
有一个 workaround 是把 null 变成 ""? 或者一个 special 字符串. 但我不确定这是不是一个比较好的实践.

Reply:
> Workaround: Replace NULL with Special Values
> 
> Using special values (e.g., “?” or “N/A”) instead of NULL is generally not a good practice because:
> 	•	It introduces ambiguity and complicates queries (e.g., SELECT queries may need to filter “special” values > explicitly).
> 	•	It undermines the semantic meaning of NULL in SQL, which represents missing or unknown data.
> 	•	It may lead to data consistency issues.

GPT 说不好

Reply:
那就还是用 id 吧, 用 long 的自增 id 应该就行了, 我觉得满不了的

Reply:
我其他的 review 里说让你换联合主键, 也可以别换了, 都用 long 的自增 id